File: framework/test2/floating/array/numeric.php
<?php
require_once 'Soda/DS/index.php';
require_once 'Soda/Unittest/index.php';

use Soda\DS;
use Soda\Unittest\TestWork;
use Soda\Unittest\Utils;

class Solution {

    /**
     * @param Double[] $x
     * @param Double[] $y
     * @return Double[]
     */
    function multiply($x, $y) {
        return array_map(fn($i) => $x[$i] * $y[$i], range(0, count($x)-1));
    }
}

$work = TestWork::create(new Solution(), 'multiply', $argv[0]);
// $work->validator = fn($e, $r) => <boolean result>
// $work->compareSerial = true;
echo $work->run(Utils::fromStdin());

File: framework/test2/floating/array/Numeric.kt
import soda.kotlin.unittest.*

class Solution {
    fun multiply(a: DoubleArray, b: DoubleArray): List<Double> {
        val res = MutableList(a.size) {0.0}
        for (i in res.indices) {
            res[i] = a[i] * b[i]
        }
        return res
    }
}

class Numeric : (String) -> String {
    override fun invoke(text: String): String {
        // * setup with your method
        val work = GenericTestWork.create(Solution()::multiply)

        // * by method that has not return value
        // val work = GenericTestWork.createVoid(Solution.METHOD_WITHOUT_RETURN)
        // * by class of data struct
        // val work = GenericTestWork.forStruct(STRUCT::class)

        // * setup validator
        // work.validator = (T, T) -> Boolean
        // work.compareSerial = true
        return work.run(text)
    }

}

fun main() {
    print(Numeric()(Utils.fromStdin()))
}
